### DISCUSSION TITLE
Deployment

### DISCUSSION THREAD

**Discussion by silvanic** [moscajs/aedes#989] [Q&A] [answered]
Hello,

I'm trying to create a simple MQTT application (broker using aedes and client, I'm using Angular). The thing is, I want to deploy it on the web to test it but I can't find any hosting service that could make it work. I'm trying Render but I keep getting “clientConnectionError” even though I don't have a connection.

Has anyone already deployed one? If so, could they kindly tell me how they did it? Thanks in advance!

**Comment by robertsLando** [comment]
for the web you have to ensure your aedes broker also listens to ws/wss protocol in order to be able to connect

**Comment by silvanic** [comment]
After a while, I finally got it to work. I had to use wss and no ws. But now I'm trying to connect via the TCP protocol without success. At least it works with WebSocket! I will investigate a bit more.

**Reply by robertsLando** [reply]
@silvanic only ws works on browser, see: https://github.com/mqttjs/MQTT.js#browser

**Reply by silvanic** [reply]
I know, but I'd like to be able to have a connection in the browser AND in another application that's not in the browser.

**Reply by robertsLando** [reply, marked as answer]
just make aedes listening both to tcp and ws so
